### PR TITLE
TSDK-221 Builders Part 2

### PR DESCRIPTION
**Pull request by DiademShoukralla**
Changes:
- Added the Mock implementations of the builder traits from the previous PR. 
- MockTransactionBuilder also has a preliminary implementation
- Replaced the parameters from being parallel lists to a list of XXBuildRequest

### CODE REVIEW

File: src/main/scala/co/topl/brambl/builders/MockOutputBuilder.scala [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package co.topl.brambl.builders
+
+import co.topl.brambl.builders.Models.OutputBuildRequest
+import co.topl.brambl.models.transaction.UnspentTransactionOutput
+
+object MockOutputBuilder extends OutputBuilder {
+  override def constructOutput(data: OutputBuildRequest): Either[BuilderError, UnspentTransactionOutput] = ???
```
Comment (mgrand-topl) [inline src/main/scala/co/topl/brambl/builders/MockOutputBuilder.scala:7]: Do you really want the mock to throw an exception?

File: src/main/scala/co/topl/brambl/builders/MockTransactionBuilder.scala [hunk 2]
```diff
@@ -0,0 +1,27 @@
+package co.topl.brambl.builders
+
+import co.topl.brambl.builders.Models.{InputBuildRequest, OutputBuildRequest}
+import co.topl.brambl.models.Datum.{IoTransaction => IoTransactionDatum}
+import co.topl.brambl.models.transaction.{IoTransaction, SpentTransactionOutput, UnspentTransactionOutput}
+
+
+object MockTransactionBuilder extends TransactionBuilder {
+  // Change needs to be explicitly added as an output.
+  // Fee is whatever is left over after the sum of the outputs is subtracted from the sum of the inputs.
```
Comment (mgrand-topl) [inline src/main/scala/co/topl/brambl/builders/MockTransactionBuilder.scala:10]: These comments would be useful in a Scaladoc for the TransactionBuilder trait

File: src/main/scala/co/topl/brambl/builders/OutputBuilder.scala [hunk 3]
```diff
@@ -1,15 +1,8 @@
 package co.topl.brambl.builders
 
-import co.topl.brambl.models.Indices
+import co.topl.brambl.builders.Models.OutputBuildRequest
 import co.topl.brambl.models.transaction.UnspentTransactionOutput
-import co.topl.brambl.models.Datum.{UnspentOutput => Datum}
-import co.topl.brambl.models.box.{Lock, Value}
 
 trait OutputBuilder {
-  def constructOutput(
```
Comment (mgrand-topl) [inline src/main/scala/co/topl/brambl/builders/OutputBuilder.scala:9]: It still needs Scaladoc

File: src/main/scala/co/topl/brambl/builders/TransactionBuilder.scala [hunk 4]
```diff
@@ -1,30 +1,20 @@
...
+import co.topl.brambl.builders.Models.{InputBuildRequest, OutputBuildRequest}
 import co.topl.brambl.models.transaction.IoTransaction
-import co.topl.brambl.models.Datum.{IoTransaction => IoTransactionDatum, SpentOutput => SpentOutputDatum, UnspentOutput => UnspentOutputDatum}
-import co.topl.brambl.models.box.Lock
+import co.topl.brambl.models.Datum.{IoTransaction => IoTransactionDatum}
 
 trait TransactionBuilder {
-  // Only considering single inputs for now so I don't have to think about box selection algorithm
   /**
-   * Construct simple transaction
```
Comment (mgrand-topl) [inline src/main/scala/co/topl/brambl/builders/TransactionBuilder.scala:11]: This needs to describe what the function does and what its parameters are for.

**Comment by DiademShoukralla** [comment]
I will address these comments in a [separate PR ](https://github.com/Topl/quivr4s/pull/17)

**Pull request closed without merging**